automatic/7zip.install/tools/chocolateyInstall.ps1

```diff
@@ -0,0 +1,40 @@
+﻿Function Get-ExplorerProcessCount
+{
+  $process = Get-Process explorer -ErrorAction SilentlyContinue
+  $processCount = ($process | Measure-Object).Count
+  return $processCount
+}
+
+$initialProcessCount = Get-ExplorerProcessCount
+Write-Warning "This installer is known to close the explorer process. This means `nyou may lose current work. `nIf it doesn't automatically restart explorer, type 'explorer' on the `ncommand shell to restart it."
+$toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+
+$filePath32 = "$toolsDir\7zip_x32.exe"
+$filePath64 = "$toolsDir\7zip_x64.exe"
+
+$filePath = if ((Get-ProcessorBits 64) -and $env:chocolateyForceX86 -ne $true) {
+  Write-Host "Installing 64 bit version" ; $filePath64
+} else { Write-Host "Installing 32 bit version" ; $filePath32 }
+
+$packageArgs = @{
+  packageName    = '7zip.install'
+  fileType       = 'exe'
+  softwareName   = '7-zip 16.04*'
+  file           = "$filePath"
+  silentArgs     = '/S'
+  validExitCodes = @(0)
+}
+
+# To prevent shimming of installers
+"" | Out-File "$filePath32.ignore"
+"" | Out-File "$filePath64.ignore"
+
+Install-ChocolateyInstallPackage @packageArgs
+
+$finalProcessCount = Get-ExplorerProcessCount
+if($initialProcessCount -lt $finalProcessCount)
+{
+  Start-Process explorer.exe
+}
+
+Remove-Item "$filePath32*","$filePath64*" -Force -ErrorAction SilentlyContinue
```

Comment:
`-ea 0` instead of `-ErrorAction SilentlyContinue` to be consistent with other packages